### PR TITLE
Update: Option to ignore for loops in init-declarations (fixes #3641)

### LINKED ISSUE #3641: Proposal: init-declarations (using never) ignore for loops
init-declarations set to never does not allow variables to be assigned on declaration, however this doesn't mix well when using a for loop. the typical use case of this setting is like so:

```
let foo;
foo = require('foo-module');
...
```

however a common use of a for loop inits the counter variable inline
```
for(var i = 0; i < bar.length; i++){ ... }
```

this raises a warning from `init-declarations: [1, "never"]`. 

ideally, i'd like to see an option present to not raise this error in some cases. e.g. inside of forloops: 
`init-declarations: [1, "never", { "ignoreForLoops": true }`

using v1.3.1

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26422076-proposal-init-declarations-using-never-ignore-for-loops?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: lib/rules/init-declarations.js [hunk 1]
```diff
@@ -49,25 +63,33 @@ module.exports = function(context) {
...
-
 };
 
 module.exports.schema = [
     {
         "enum": ["always", "never"]
+    },
+    {
+        "type": "object",
+        "properties": {
```
Comment (ilyavolodin) [inline lib/rules/init-declarations.js]: I think this should only be allowed for `never`, but shouldn't be allowed for `always`.

Comment (alberto) [inline lib/rules/init-declarations.js]: Is there any similar rule I can use as a reference? I couldn't find any, and I'm not sure what I am supposed to do.

Comment (ilyavolodin) [inline lib/rules/init-declarations.js]: @btmills Any ideas?

Comment (btmills) [inline lib/rules/init-declarations.js]:
```json
{
    "anyOf": [
        {
            "type": "array",
            "items": [
                {
                    "enum": [0, 1, 2]
                },
                {
                    "enum": ["always"]
                }
            ],
            "minItems": 1,
            "maxItems": 2
        },
        {
            "type": "array",
            "items": [
                {
                    "enum": [0, 1, 2]
                },
                {
                    "enum": ["never"]
                },
                {
                    "type": "object",
                    "properties": {
                        "ignoreForLoops": {
                            "type": "boolean"
                        }
                    },
                    "additionalProperties": false
                }
            ],
            "minItems": 1,
            "maxItems": 3
        }
    ]
}
```

Comment (alberto) [inline lib/rules/init-declarations.js]: Thanks! That's json-schema, right?

Comment (ilyavolodin) [inline lib/rules/init-declarations.js]: Right, that would go into `module.exports.schema`

Comment (alberto) [inline lib/rules/init-declarations.js]: Yes, I already updated it. :)

**Pull request merged**